File: templates/Clojure/procon.clj
(defn main
  "Main function"
  []
  (loop [line (read-line)]
    (if (nil? line)
      nil
      (do (let [tokens (map #(Long/parseLong %) (.split line " +"))]
            (println tokens)
            <+CURSOR+>)
          (recur (read-line))))))

(main)

File: templates/Lua/procon.lua
#!/usr/bin/env lua

if ... then
  module(..., package.seeall)
end


function split(str, delim)
  if string.find(str, delim) == nil then
    return { str }
  end
  local result = {}
  local pat = '(.-)' .. delim .. '()'
  local lastPos
  for part, pos in string.gfind(str, pat) do
    table.insert(result, part)
    lastPos = pos
  end
  table.insert(result, string.sub(str, lastPos))
  return result
end


if not ... then
  line = io.read()
  while line do
    tokens = split(line, ' +')
    <+CURSOR+>

    line = io.read()
  end
end
